src/tests/ftest/pool/dynamic_server_pool.py

```diff
@@ -0,0 +1,156 @@
+    Then we'll create one pool on this new server and another one over all three
+    servers. This covers SRS-10-0054.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DynamicServerPool object."""
+        super(DynamicServerPool, self).__init__(*args, **kwargs)
+        self.expected_uuids = []
+        self.uuid_to_ranks = {}
+
+    def setUp(self):
+        super(DynamicServerPool, self).setUp()
+        self.pool = []
+
+    def verify_uuids(self):
+        """Call dmg pool list to get the list of UUIDs and verify against the
+        expected UUIDs.
+        """
+        actual_uuids = []
+        output_list = self.get_dmg_command().get_output("pool_list")
+        for uuid_sr in output_list:
+            actual_uuids.append(uuid_sr[0])
+        actual_uuids.sort()
+        self.expected_uuids.sort()
+        self.assertEqual(self.expected_uuids, actual_uuids)
+
+    def check_pool_location(self, hosts, uuid_to_ranks):
+        """Iterate pools and verify that the UUID-named directory exists in
+        each hosts.
+
+        Args:
+            hosts (list of str): Hostnames to search the UUID-named directory.
+            uuid_to_ranks (str to list of int dictionary): UUID to rank list
+                dictionary.
+        """
+        RC_SUCCESS = 0
+        errors = []
+        for pool in self.pool:
+            # Note that we don't check mapping between rank and hostname, but it
+            # appears that self.hostlist_servers[0] is always rank0, 1 is rank1,
+            # and the extra server we'll be adding will be rank2.
+            for rank, host in enumerate(hosts):
+                rc = check_for_pool(host, pool.uuid.lower())
+                pool_exists_on_host = rc == RC_SUCCESS
+                # If this rank is in the rank list, there should be the
+                # UUID-named directory; i.e., pool_exist_on_host is True.
+                pool_expected = rank in uuid_to_ranks[pool.uuid.lower()]
+                if pool_expected != pool_exists_on_host:
+                    error_message = "Is the pool expected to exist? " +\
+                        str(pool_expected) + "; Actual: " +\
+                        str(pool_exists_on_host)
+                    errors.append(error_message)
+        self.assertEqual(len(errors), 0, "\n".join(errors))
+
+    def create_pool_with_ranks(self, ranks, tl_update=False):
+        """Create a pool with possibly --ranks parameter.
+
+        Args:
+            ranks (list of int): --ranks value.
+            tl_update (bool, optional): Whether to update target_list, which
+                means pass in value to --ranks during pool create. Defaults to
+                False.
+        """
+        if tl_update:
+            self.pool.append(self.get_pool(create=False))
+            self.pool[-1].target_list.update(ranks, "pool.target_list")
+        else:
+            self.pool.append(self.get_pool())
+        self.pool[-1].create()
```

Comment:
Since `self.get_pool()` is being called w/o `create=False` in the `else` case, the pool will have already been created and the call to `self.pool[-1].create()` in that situation is unnecessary.  You could move the `self.pool[-1].create()` call like so or just include `create=False` in both `self.get_pool()` calls.

It may not make a difference, but calling `self.pool.append(self.get_pool(create=False)); self.pool[-1].create()` is not the same as `self.pool.append(self.get_pool())`.  The equivalent call would be `self.pool.append(self.get_pool(connect=False))`.  I've been thinking lately that `connect=False` should probably be the default - which would alleviate this inconsistency.  That would be a bit more work and outside the scope of this PR. 
  
```suggestion
        if tl_update:
            self.pool.append(self.get_pool(create=False))
            self.pool[-1].target_list.update(ranks, "pool.target_list")
            self.pool[-1].create()
        else:
            self.pool.append(self.get_pool())
```

Reply:
Done. Thanks for catching. Yeah it looks like it was destroying the pool before creating the second pool. It was working, but unnecessary.